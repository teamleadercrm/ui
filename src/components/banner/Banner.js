import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Box from '../box';
import Island from '../island';
import Section from '../section';
import IconButton from '../iconButton';
import theme from './theme.css';
import { IconCloseMediumOutline } from '@teamleader/ui-icons';

/** @type {React.ComponentType<any>} */
class Banner extends PureComponent {
  getCloseButtonColor() {
    const { color } = this.props;
    return color === 'white' ? 'neutral' : color;
  }

  render() {
    const { children, className, icon, onClose, fullWidth, ...others } = this.props;
    const Element = fullWidth ? Section : Island;

    return (
      <Element data-teamleader-ui="banner" className={className} {...others}>
        <div className={theme['inner']}>
          {icon && <span className={theme['icon']}>{icon}</span>}
          <Box flex={1} element="span" paddingRight={onClose && 7}>
            {children}
          </Box>
          {onClose && (
            <IconButton
              className={theme['close-button']}
              icon={<IconCloseMediumOutline />}
              color={this.getCloseButtonColor()}
              onClick={onClose}
            />
          )}
        </div>
      </Element>
    );
  }
}

Banner.propTypes = {
  /** The content to display inside the banner. */
  children: PropTypes.node,
  /** A class name for the wrapper to give custom styles. */
  className: PropTypes.string,
  /** The color of the banner. */
  color: PropTypes.oneOf(['white', 'neutral', 'mint', 'violet', 'ruby', 'gold', 'aqua']),
  /** The icon displayed on the left inside the banner. */
  icon: PropTypes.element,
  /** Callback function that is fired when the close button of the banner is clicked. */
  onClose: PropTypes.func,
  /** If true, component will take the full width of it's container. */
  fullWidth: PropTypes.bool,
};

Banner.defaultProps = {
  color: 'white',
};

export default Banner;
