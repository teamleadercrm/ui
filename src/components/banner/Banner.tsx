import React from 'react';

import Box from '../box';
import Island from '../island';
import Section from '../section';
import IconButton from '../iconButton';
import theme from './theme.css';
import { IconCloseMediumOutline } from '@teamleader/ui-icons';

export interface BannerProps {
  /** The content to display inside the banner. */
  children?: React.ReactNode;
  /** A class name for the wrapper to give custom styles. */
  className?: string;
  /** The color of the banner. */
  color?: 'white' | 'neutral' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua';
  /** The size of the banner. */
  size?: 'small' | 'medium' | 'large';
  /** The icon displayed on the left inside the banner. */
  icon?: React.ReactElement;
  /** Callback function that is fired when the close button of the banner is clicked. */
  onClose?(...args: unknown[]): unknown;
  /** If true, component will take the full width of it's container. */
  fullWidth?: boolean;
}

const Banner = ({ children, className, color = 'white', icon, onClose, fullWidth, ...others }: BannerProps) => {
  const Element = fullWidth ? Section : Island;

  return (
    <Element data-teamleader-ui="banner" className={className} {...others}>
      <div className={theme['inner']}>
        {icon && <span className={theme['icon']}>{icon}</span>}
        <Box flex={1} element="span" paddingRight={onClose && 7}>
          {children}
        </Box>
        {onClose && (
          <IconButton
            className={theme['close-button']}
            icon={<IconCloseMediumOutline />}
            color={color === 'white' ? 'neutral' : color}
            onClick={onClose}
          />
        )}
      </div>
    </Element>
  );
};

export default Banner;
