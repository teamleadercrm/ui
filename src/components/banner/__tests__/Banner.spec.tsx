import React from 'react';
import { render } from '@testing-library/react';
import Banner from '../Banner';

describe('Component - Banner', () => {
  it('renders', async () => {
    const { asFragment } = render(<Banner>Foobar</Banner>);
    expect(asFragment()).toMatchSnapshot();
  });
});
