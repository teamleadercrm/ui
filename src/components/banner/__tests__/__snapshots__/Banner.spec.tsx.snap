// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Banner renders 1`] = `
<DocumentFragment>
  <div
    class="box padding-bottom-4 padding-left-4 padding-right-4 padding-top-4 white banner"
    color="white"
    data-teamleader-ui="banner"
    style="border-radius: 4px; border-top-left-radius: 4px; border-top-right-radius: 4px; border-bottom-left-radius: 4px; border-bottom-right-radius: 4px;"
  >
    <div
      class="inner"
    >
      <span
        class="box"
        style="flex: 1;"
      >
        Foobar
      </span>
    </div>
  </div>
</DocumentFragment>
`;
